internal/provider/resource_gitlab_group_saml_link.go

```diff
@@ -0,0 +1,163 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_group_saml_link", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_group_saml_link`" + ` resource allows to manage the lifecycle of an SAML integration with a group.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/groups.html#saml-group-links)`,
+
+		CreateContext: resourceGitlabGroupSamlLinkCreate,
+		ReadContext:   resourceGitlabGroupSamlLinkRead,
+		DeleteContext: resourceGitlabGroupSamlLinkDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceGitlabGroupSamlLinkImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"group_id": {
```

Comment:
The upstream API supports both, the group as integer id and full path. I think it would make more sense to have a group attribute like this:

```go
"group": {
		Description: "The ID or URL-encoded path of the group.",
		Type:        schema.TypeString,
		Required:    true,
		ForceNew:    true,
},
```

There are other resources supported this already, like the `gitlab_group_access_token` resource:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/630a88e620f604322d8e28cc8ed7f16558ecb858/internal/provider/resource_gitlab_group_access_token.go#L48